assets/src/js/admin/reports/components/rest-list/index.js

```diff
@@ -0,0 +1,93 @@
+// Vendor dependencies
+import axios from 'axios'
+import { useState, useEffect, Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+// Components
+import List from '../list'
+import DonorItem from '../donor-item'
+import LocationItem from '../location-item'
+import DonationItem from '../donation-item'
+
+// Store-related dependencies
+import { useStoreValue } from '../../app/store'
+
+const RESTList = ({endpoint}) => {
+
+	// Use period from store
+	const [{ period }, dispatch] = useStoreValue()
+
+	// Use state to hold data fetched from API
+	const [fetched, setFetched] = useState(null)
+
+	// Fetch new data and update List when period changes
+	useEffect(() => {
+		axios.get(wpApiSettings.root + 'give-api/v2/reports/' + endpoint, {
```

Comment:
@henryholtgeerts I recommend creating a helper function to build API URL dynamically instead of hardcode them in code.
I am open to discussion for a data model for a helper function. 
My recommendation is:
1. create helper function with name `giveGetAPIURL`
2. Allow this function to  accept data model:
```
{
version: 'v2' \\ required.
type:  'report' \\ required.
query: {endpoint} \\  required. you can mention endpoint here line donations, donors etc.

}
```

Reply:
Because Reports is currently being built with more of an MVP approach, I opted to hardcode these URLs. The API is only called directly in 3 components (RESTChart, RESTMiniChart, RESTList). I'm open to introducing this helper function, but I'm curious what advantages are really offered by that abstraction. I agree with the sentiment of wanting to avoid hardcoding things, but ultimately the helper function just might not be that helpful in this case, since we only call the API directly in 3 places. Open to discussion on this! 👍 @JasonTheAdams

Reply:
@henryholtgeerts If we shall develop version 3 for API then we have to update URL three or many places but if we will provide abstraction then we only have to change at one place.

Reply:
Ahh...I undertand. So the accepted data modal would look more like: 
```
{
version: 'v2' \\ NOT required (defaults to v2)
type:  'report' \\ required.
query: {endpoint} \\  required. you can mention endpoint here line donations, donors etc.
}
```
☝️ Assuming we would still like the option to control which API version the URL returns with, although there could be a default (like v2).

I agree that this could become a headache if many more components are added which directly call the API, however for right now I am going to suggest we leave the PR as is, and I will open an issue to create/implement this helper function down the road. It feels beyond the scope of the issue this PR addresses, and I think we can set it up once this MVP is together.

Reply:
@henryholtgeerts Agree, create an issue for this